### DISCUSSION TITLE
How to throw ArgumentNullException instead of returning default when source is null

### DISCUSSION THREAD

**Discussion by gpetrou** [riok/mapperly#1378] [General]
Currently, I see in the generated code the first check is:
```csharp
if (source == null)
    return default;
```
Is it possible to configure this to have the following instead?
```csharp
if (source == null)
    throw new ArgumentNullException(nameof(source));
```

**Comment by latonz** [comment]
Are your classes / your mapper definition in a nullable disabled or enabled context?

If only your parameter is nullable but your return type is not nullable, throwing is the default behaviour. It can be customized with the `ThrowOnMappingNullMismatch` property on the `MapperAttribute`. The customization is not possible if the return type is also nullable.
